web/cobrands/fixmystreet/admin.js

```diff
@@ -123,5 +123,93 @@ $(function(){
             }
         }
     });
+
+    // Bits for the report extra fields form builder:
+
+    // Some elements' visibility depends on JS being available
+    $(".js-metadata-items .js-hidden").hide();
+    $(".js-metadata-items .js-visible").show();
+
+    // If type is changed to 'singlevaluelist' show the options list
+    $(".js-metadata-items").on("change", ".js-metadata-item-type", function() {
+        var $this = $(this);
+        var shown = $this.val() === 'singlevaluelist';
+        var $list = $this.closest(".js-metadata-item").find('.js-metadata-options');
+        $list.toggle(shown);
+    });
+    // call immediately to perform page setup
+    $(".js-metadata-item-type").change();
+
+    // Options can be removed by clicking the 'remove' button
+    $(".js-metadata-items").on("click", ".js-metadata-option-remove", function(e) {
+        e.preventDefault();
+        var $this = $(this);
+        var $item = $this.closest(".js-metadata-item");
+        $this.closest('li').remove();
+        renumber_metadata_options($item);
```

Comment:
Good point, you're right that it would be simpler to just increment the maximum. However the `data-field-name` stuff is still required cos you still need to update the names of fields in the newly added item.